TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -150,9 +150,25 @@ public override bool Visit(Section section)
             return true;
         }
 
-        public override bool Visit(Set setStatement)
+        public override bool Visit(Set set)
         {
-            SetStatementChecker.CheckStatement(setStatement);
+            if (set.StorageAreaWritesDataDefinition != null && set.CodeElement is SetStatementForConditions setConditions && setConditions.IsSendingValueFalse)
+            {
+                // Statement is a SET TO FALSE: we need to check that all conditions have been defined with a clause WHEN SET TO FALSE
+                foreach (var condition in setConditions.Conditions)
+                {
+                    if (condition.StorageArea != null)
+                    {
+                        set.StorageAreaWritesDataDefinition.TryGetValue(condition.StorageArea, out var dataCondition);
```

Comment:
no test on return value of this `TryGetValue` ??? 👿
Ok this is written this way multiple times in the file and there is also `GetDataDefinitionFromStorageAreaDictionary` method which acts as a helper to do that. So I'm not sure anymore

Reply:
`GetDataDefinitionFromStorageAreaDictionary` is indeed a good option.
However when used in a loop it will test in each call that dictionary is not null.
It is not necessary here.

So I keep `TryGetValue `and check its return value.